extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerImpl.java

```diff
@@ -183,7 +184,10 @@ public WorkerImpl(MSQWorkerTask task, WorkerContext context)
     this.context = context;
     this.selfDruidNode = context.selfNode();
     this.processorBouncer = context.processorBouncer();
-    this.durableStageStorageEnabled = MultiStageQueryContext.isDurableStorageEnabled(
+    this.durableStageStorageEnabled = MultiStageQueryContext.isDurableShuffleStorageEnabled(
+        QueryContext.of(task.getContext())
+    );
+    this.durableTaskIntermediateStorageEnabled = MultiStageQueryContext.isDurableTaskIntermediateStorageEnabled(
```

Comment:
I think for this parameter to get populated from the user query, we would need to pass it explicitly in the `MSQWorkerTaskLauncher` class. There should be a preexisting example of `CTX_ENABLE_DURABLE_SHUFFLE_STORAGE` in there already.

Reply:
not applicable now since dropped the flag itself